benchmark/buffers/buffer-bytelength-utf8.js

```diff
@@ -0,0 +1,42 @@
+var common = require('../common');
+
+var bench = common.createBenchmark(main, {
+    len: [4, 16, 64, 256, 1024],
+    n: [1e6]
+});
+
+// 16 chars of each byte length
+var chars = {
+  0: 'hello brendan!!!', // 1 byte
+  1: 'ΰαβγδεζηθικλμνξο', // 2 bytes
+  2: '挰挱挲挳挴挵挶挷挸挹挺挻挼挽挾挿', // 3 bytes
+  3: '𠜎𠜱𠝹𠱓𠱸𠲖𠳏𠳕𠴕𠵼𠵿𠸎𠸏𠹷𠺝𠺢' // 4 bytes
+};
+
+function main(conf) {
+  var n = conf.n | 0;
+  var len = conf.len | 0;
+
+  // make a string with them all mixed together
+  var str = buildStr(len);
+
+  bench.start();
+  for (var i = 0; i < n; i++) {
+    Buffer.byteLength(chars[0], 'utf8');
+    Buffer.byteLength(chars[1], 'utf8');
+    Buffer.byteLength(chars[2], 'utf8');
+    Buffer.byteLength(chars[3], 'utf8');
+    Buffer.byteLength(str, 'utf8');
```

Comment:
I think there's more `Buffer.byteLength()` calls than necessary. What I had in mind for my original comment was something more like:

```javascript
Buffer.byteLength(chars[i % 4], 'utf8');
```

Additionally, it might be a good idea to increase the "default" `n` value to something that causes the benchmark not to exit almost immediately, if it's doing that currently. The longer it runs, the more stable numbers you will get because v8 will have had sufficient time to do compilations and performed any optimizations after the JIT has analyzed the function and whatnot.